drizzlepac/haputils/product.py

```diff
@@ -161,6 +161,21 @@ def __init__(self, prop_id, obset_id, instrument, detector, filename, filetype,
 
         log.debug("Total detection object {}/{} created.".format(self.instrument, self.detector))
 
+    def update_drizpars(self):
+        """ Update ALL products with final name of trailer file """
+        # Update this total product
+        fits.setval(self.drizzle_filename, 'DRIZPARS', value=self.trl_filename)
+
+        # Update all filter drizzle products that were actually written out
+        for fdp in self.fdp_list:
+            if os.path.exists(fdp.drizzle_filename):
+                fits.setval(fdp.drizzle_filename, 'DRIZPARS', value=fdp.trl_filename)
+
+        # Update all exposure drizzle products that were actually written out
+        for edp in self.edp_list:
+            if os.path.exists(edp.drizzle_filename):
+                fits.setval(edp.drizzle_filename, 'DRIZPARS', value=edp.trl_filename)
```

Comment:
Curious why there is a check for the individual exposure product object, but not for the filter product?

Reply:
Lines 170-172 adds the keyword to the filter products with Line 171 checking to see that it exists in the first place.  This seems to address your question.